webapp/src/components/BidAssetPage/BidEstatePage/BidEstatePage.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Container, Message, Loader } from 'semantic-ui-react'
+import { t, T } from '@dapps/modules/translation/utils'
+
+import { locations } from 'locations'
+import { isOpen } from 'shared/listing'
+import BidAssetForm from '../BidAssetForm'
+import { authorizationType, bidType } from 'components/types'
+import Estate from 'components/Estate'
+import EstateName from 'components/EstateName'
+import TxStatus from 'components/TxStatus'
+import EstateModal from 'components/EditEstatePage/EditEstateMetadata/EstateModal'
+
+export default class BidEstatePage extends React.PureComponent {
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    bid: bidType,
+    authorization: authorizationType,
+    isLoading: PropTypes.bool.isRequired,
+    isTxIdle: PropTypes.bool.isRequired,
+    onBid: PropTypes.func.isRequired,
+    onCancel: PropTypes.func.isRequired
+  }
+
+  renderLoading() {
+    return (
+      <div>
+        <Loader active size="massive" />
+      </div>
+    )
+  }
+
+  render() {
+    const {
+      id,
+      bid,
+      isLoading,
+      authorization,
+      isTxIdle,
+      onBid,
+      onCancel
+    } = this.props
+
+    const bidIsOpen = isOpen(bid)
+
+    if (isLoading) {
+      return this.renderLoading()
+    }
+
+    return (
+      <Estate id={id} ownerNotAllowed>
+        {estate => {
+          const isERC721BidAllowed =
```

Comment:
Suggestion: use implicit return to avoid nested `return` statements like the one below, and extract the variable declarations like this one to a class method that can be called from within the implicit return, like `this.isAllowed()` and then having 

```js
isAllowed = () => {
  return this.props.authorization && this.props.authorization.allowances.ERC721Bid.MANAToken
}
```